lib/common/utils.ts

```diff
@@ -280,9 +280,15 @@ export function makeZoneAwareRemoveAllListeners(fnName: string, useCapturingPara
   const defaultUseCapturing = useCapturingParam ? false : undefined;
 
   return function zoneAwareRemoveAllListener(self: any, args: any[]) {
+    var target = self || _global;
+    if (args.length === 0) {
+      // remove all listeners without eventName
+      target[EVENT_TASKS] = [];
+      target[symbol]();
+      return;
+    }
     var eventName = args[0];
     var useCapturing = args[1] || defaultUseCapturing;
-    var target = self || _global;
     var eventTasks = findAllExistingRegisteredTasks(target, eventName, useCapturing, true);
```

Comment:
the cancelTask will just call the original EventEmitter.removeListener,  I think we don't need to cancel it, because when we call target[symbol]() https://github.com/angular/zone.js/pull/516/files#diff-f5e084bcb7c43d977d56c3b791bbca2dR287, the EventEmitter.removeAllListeners will do it for us.

and the cancelTask call here is also not necessary either I think, https://github.com/angular/zone.js/blob/master/lib/common/utils.ts#L290